rfc/2022/2022-09-15_Design-System-Team_Migrate-vets-website-off-of-formation.md

```diff
@@ -0,0 +1,106 @@
+In order to continue to receive updates from USWDS, we need to switch to a more recent version.
+
+## Motivation
+
+The Design System team is developing a new CSS library which is based on USWDS3 and may contain additional VA-specific styling.
+By relying on USWDS internals, we are able to provide a lot of utilities while having to maintain less SASS ourselves.
+We also want to namespace all of the classes we provide rather than allow developers full unlimited access to everything within USWDS.
+This gives the Design System more control and will also make future maintenance and updates easier.
+
+We also hope to be able to improve performance with this new css library in a number of ways, including:
+
+- smaller stylesheets due to elimination of `!important` declaration
+- smaller stylesheets due to simpler utility names (`vads-u-background-color--gray` => `vads-bg-gray`)
+
+(Note: these name changes are why teams will need to update their apps to use the names of the new classes)
+
+We'd also like to explore the possibility of allowing teams to rely less on global site-wide assets and declare a minimum subset of which DS assets they will need. Much of this work is out of scope for this RFC.
+
+## Design
+
+Before this migration can begin there are prerequisites which must happen.
+
+### Prerequisites
+
+1. Application, sitewide, & platform code will need to be using proper VADS web components. This means no React components,
+and no CSS class components as these rely on `formation` components.
+    - This will unblock those modules from being removed in formation
+1. We will need a published version of the `css-library` package
+1. Write a new ESLint rule which can trigger when a `vads-u-*` class is used.
+
+Once these have been met we can move on to beginning the migration
+
+### Migration
+
+In order to begin migrating we will need to make the `css-library` available, and this can be done in one of two ways.
+
+1. Add the stylesheets to the sitewide SASS file.
+    - This will cause every page to download the file whether they use the utilities or not
+    - Could be a "final state" of the migration
+1. Add the stylesheets to each specific app's SASS file as they are migrated.
+    - This will bloat the app's bundle size, but other pages and apps won't be affected
+
+We will be starting with migrating the utility classes, and the css library will export a `utilities.css` file as part of its build which can be imported in either location. All of the steps outlined here will begin with a single app, and then expanded to other apps once the pilot app has been successfully migrated.
+
+#### Remove old VADS utility classes from markup
+We will first identify a team and/or app and work with them on migrating their markup to use the new USWDS3-based utilities provided by `css-library`. As part of this we should also see if we can eliminate any unused CSS from the application's SASS file.
```

Comment:
I believe we'd also use this as an opportunity to build and refine ESLint rules that would make it easy for subsequent teams to convert to the new CSS library, correct? Perhaps more of a secondary goal as we could iterate with future teams on this in Q1 of next year but, might be worth mentioning.

Reply:
Are you referring to the prerequisite?

https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/blob/64eb3d4245eb71bdb2ac706cd51fcc626d4825ff/rfc/2022/2022-09-15_Design-System-Team_Migrate-vets-website-off-of-formation.md?plain=1#L42

Reply:
Yes. Perhaps just change "write" to "write and refine".

Reply:
https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/pull/48/commits/29183872dcad32a7a600549a74d9f8d3851f24ce